src/H5Fint.c

```diff
@@ -1333,9 +1333,12 @@ H5F__new(H5F_shared_t *shared, unsigned flags, hid_t fcpl_id, hid_t fapl_id, H5F
         }
 
         /* Free VOL object */
-        if (f->vol_obj)
+        if (f->vol_obj) {
+            if (NULL == H5VL_object_unwrap(f->vol_obj))
```

Comment:
I believe that this is the right solution here. While the call to H5VL_object_unwrap returns a VOL object that just gets discarded here, it is still important to make sure the VOL object gets unwrapped. It seems this is a one-off case that requires this specifically. Putting the unwrap logic into H5VL_free_object seems inappropriate since every usage I can find in the library seems to already be dealing with an unwrapped object, so the unwrap logic would just be unnecessary overhead.

Reply:
This code is already surrounded by an existing `if (f->vol_obj)` check though, so we know that it would be non-NULL if it reaches here. Like you mention though, f->vol_obj will probably never be non-NULL here, so it's probably unnecessary to do the VOL object unwrapping during H5F__new. I can revert that back to the existing code, even though that probably will never be reached either. It's definitely needed inside H5F__dest though.

Reply:
OK, cool.   Maybe take this out and replace it with "HDassert(NULL == f->vol_obj);" ?

Reply:
Done.